sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -108,7 +94,8 @@ static PipelineOptions getDefaultConfig(String scheme) {
   static FileSystem getFileSystemInternal(URI uri) {
     String lowerCaseScheme = (uri.getScheme() != null
         ? uri.getScheme().toLowerCase() : LocalFileSystemRegistrar.LOCAL_FILE_SCHEME);
-    return getRegistrarInternal(lowerCaseScheme).fromOptions(getDefaultConfig(lowerCaseScheme));
+    return getRegistrarInternal(lowerCaseScheme)
+        .fromOptions(checkNotNull(defaultConfig, "defaultConfig"));
```

Comment:
Should the error message be improved? It seems like if this fails the resolution is complex. Like a runner should have called `FileSystems.setDefaultConfigInWorkers`? Or the user needs to register a config for the specific URI?

Probably the URI should be in the error message too.